tokio-console/src/state/tasks.rs

```diff
@@ -217,16 +221,35 @@ impl TasksState {
                     warnings: Vec::new(),
                     location,
                 };
-                task.lint(linters);
+                if matches!(task.lint(linters), TaskLintResult::RequiresRecheck) {
+                    next_pending_lint.insert(task.id);
+                }
                 Some((id, task))
             });
 
         for (stats, mut task) in self.tasks.updated(stats_update) {
             tracing::trace!(?task, ?stats, "processing stats update for");
             task.stats = stats.into();
-            task.lint(linters);
+            if matches!(task.lint(linters), TaskLintResult::RequiresRecheck) {
+                next_pending_lint.insert(task.id);
+            } else {
+                // Avoid linting this task again this cycle
+                self.pending_lint.remove(&task.id);
+            }
```

Comment:
nit: It's more idiomatic Rust to use a match statement here (even more so than using `if let` in the comment above):
```suggestion
            match task.lint(linters) {
                TaskLintResult::RequiresRecheck => next_pending_lint.insert(task.id),
                // Avoid linting this task again this cycle
                _ => self.pending_lint.remove(&task.id),
            }
```